Geometry_Engine/Compute/SortAlongCurve.cs

```diff
@@ -69,6 +83,42 @@ public static List<Point> SortAlongCurve(this List<Point> points, Circle circle,
 
         /***************************************************/
 
+        [Description("Sorts a set of points along the provided curve. Sorting is done based on the curve parameter at the position of the closest point on the curve in relation to each provided Point.")]
+        [Input("points", "The list of points to sort.")]
+        [Input("ellipse", "The Ellipse to sort along.")]
+        [Input("distanceTolerance", "Tolerance to be used for distance computations in the method.", typeof(Length))]
+        [Input("angleTolerance", "Tolerance to be used for angle computations in the methods.", typeof(Angle))]
+        [Output("sortedPts", "The provided points sorted along the curve.")]
+        public static List<Point> SortAlongCurve(this List<Point> points, Ellipse ellipse, double distanceTolerance = Tolerance.Distance, double angleTolerance = Tolerance.Angle)
+        {
+            if(points == null || points.Count == 0)
+                return new List<Point>();
+
+            if (ellipse.IsNull())
+                return new List<Point>();
```

Comment:
Slightly less sure here, but could argue it both ways really. Happy to change to null return if you feel strongly about it though :)

Reply:
I would return `null` - from what I understand, in case of sorting the input objects should never change, only their order. So if it is impossible to deduct the valid order, it does not mean the points disappear